prepare-install.js

```diff
@@ -0,0 +1,50 @@
+const { unlinkSync, readFileSync, writeFileSync } = require('fs');
+const { join } = require('path');
+
+const pkgJson = readFileSync(join(__dirname, 'package.json'), 'utf8');
+const pkg = JSON.parse(pkgJson);
+const isWin = process.platform === 'win32';
+const isNode12 = process.version.startsWith('v12.');
+
+function saveExact(pkg, name) {
+  const value = pkg.devDependencies[name];
+  if (value && value.startsWith('^')) {
+    pkg.devDependencies[name] = value.slice(1);
+  }
+}
+
+if (isWin || isNode12) {
+  unlinkSync(join(__dirname, 'yarn.lock'));
+  // Delete the integration tests that fail in both Windows and Node12
+  unlinkSync(join(__dirname, 'test', 'integration', 'yoga-layout.js'));
+  delete pkg.devDependencies['yoga-layout'];
+
+  // Since we are going to regenerate yarn.lock, lets make sure to pick the exact version
+  saveExact(pkg, 'ts-loader');
+  saveExact(pkg, 'tsconfig-paths');
+  saveExact(pkg, 'tsconfig-paths-webpack-plugin');
+  saveExact(pkg, 'typescript');
```

Comment:
@guybedford Any idea why typescript isn't working with Node 12?

I tried pinning these versions but I still get `Callback was already called`.

```
node dist/ncc/cli.js build test/integration/test.ts -o tmp
ncc: Version 0.21.1
ncc: Compiling file index.js
ncc: Using typescript@3.2.CUSTOM (local user-provided)
(node:86325) UnhandledPromiseRejectionWarning: Error: Callback was already called.
    at throwError (evalmachine.<anonymous>:1:479349)
    at evalmachine.<anonymous>:1:488552
    at evalmachine.<anonymous>:1:713553
    at evalmachine.<anonymous>:1:1073616
    at evalmachine.<anonymous>:1:1073256
    at evalmachine.<anonymous>:1:1373913
    at evalmachine.<anonymous>:1:1073645
    at eval (eval at create (evalmachine.<anonymous>:1:278698), <anonymous>:30:1)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(node:86325) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:86325) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Reply:
Regenerating the entire yarn.lock file seems pretty brave... I ran this same test with the original yarn.lock and it works fine, so it must be some dependency or transitive dependency issue.

Digging into the bug above it seems to be caused by this error:

```
TypeError: Cannot read property 'getOptionsDiagnostics' of undefined
    at Object.getCompilerOptionsDiagnostics (/home/guybedford/Projects/ncc/test/integration/node_modules/typescript/index.js:1:2941274)
    at provideCompilerOptionDiagnosticErrorsToWebpack (/home/guybedford/Projects/ncc/node_modules/ts-loader/dist/after-compile.js:39:31)
    at /home/guybedford/Projects/ncc/node_modules/ts-loader/dist/after-compile.js:17:9
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/guybedford/Projects/ncc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:14:1)
    at AsyncSeriesHook.lazyCompileHook (/home/guybedford/Projects/ncc/node_modules/tapable/lib/Hook.js:156:20)
    at /home/guybedford/Projects/ncc/node_modules/webpack/lib/Compiler.js:682:32
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/guybedford/Projects/ncc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:4:1)
    at AsyncSeriesHook.lazyCompileHook (/home/guybedford/Projects/ncc/node_modules/tapable/lib/Hook.js:156:20)
    at /home/guybedford/Projects/ncc/node_modules/webpack/lib/Compilation.js:1359:36
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/guybedford/Projects/ncc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:4:1)
```

Perhaps you can try make more gradual changes to the Yarn.lock instead of this full regeneration to achieve this PR?

Reply:
Oddly enough, upgrading the dependencies breaks Node 10 for leveldown.

>  Error: No native build was found for platform=linux arch=x64 runtime=node abi=64 uv=1 libc=glibc [see logs](https://github.com/zeit/ncc/pull/522/checks?check_run_id=534552942)

Node 12 fails on the same test as before:

> ncc build test/integration/test.ts -o tmp